### PR TITLE
devtools: writing to filesystem in FILE thread

### PR DESCRIPTION
**Pull request by deepak1556**

### CODE REVIEW

**Comment by zcbenz** [comment]
`DevToolsAPI.appendedToURL` and `DevToolsAPI.savedURL` should be called after the file writing is done.

**Comment by deepak1556** [comment]
Arent they independent calls ?

**Comment by zcbenz** [comment]
`DevToolsAPI.appendedToURL` is used to inform the devtools that writing has been done, if we call it before writing is done then it is possible that devtools would start writing to the same file while previous writing is not finished.

**Comment by deepak1556** [comment]
Fixed, thanks for the clarification!

File: atom/browser/common_web_contents_delegate.cc [hunk 1]
```diff
@@ -86,6 +87,35 @@ base::DictionaryValue* CreateFileSystemValue(const FileSystem& file_system) {
...
+void AppendToFile(brightray::InspectableWebContents* web_contents,
+                  const base::FilePath& path,
+                  const std::string& content,
+                  const std::string& url) {
+  DCHECK_CURRENTLY_ON(content::BrowserThread::FILE);
+  DCHECK(!path.empty());
+
+  if (base::AppendToFile(path, content.data(), content.size())) {
+    base::StringValue url_value(url);
+    web_contents->CallClientFunction(
```
Comment (zcbenz) [inline atom/browser/common_web_contents_delegate.cc]: You can not execute JavaScript in FILE thread.

Comment (deepak1556) [inline atom/browser/common_web_contents_delegate.cc]: should it be handled with `postTaskAndReplyWithResult` ?

Comment (zcbenz) [inline atom/browser/common_web_contents_delegate.cc]: `PostTaskAndReply` should be enough, `DevToolsAPI.savedURL` doesn't care whether file is written successfully.

Comment (deepak1556) [inline atom/browser/common_web_contents_delegate.cc]: Fixed, thanks!

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**